src/stackdriver-nozzle/stackdriver/metrics_buffer.go

```diff
@@ -16,12 +16,83 @@
 
 package stackdriver
 
-import "reflect"
+import (
+	"context"
+	"github.com/mitchellh/hashstructure"
+	"reflect"
+	"sync"
+	"time"
+)
 
 type MetricsBuffer interface {
 	PostMetric(*Metric)
```

Comment:
Suggest explicitly checking over a random wait period. Perhaps bock on buffer.HasMetrics() with a timeout. Just gives some more certainty and prevents any unneeded sleeping.